test/middleware.js

```diff
@@ -0,0 +1,24 @@
+'use strict'
+/* global describe, it */
+
+const middleware = require('../lib/middleware')
+
+require('chai').should()
+
+describe('middleware', () => {
+  it('should add a list of callbacks to global middleware', () => {
+    const globalMiddleware = []
+
+    middleware(globalMiddleware, ['callback1', 'callback2'])
+
+    globalMiddleware.should.have.lengthOf(2)
+  })
+
+  it('should add a single callback to global middleware', () => {
```

Comment:
mind adding one functional test that demonstrates global middleware being applied? either in this file, or perhaps in the `yargs.js` test file.